test/augmentation/test_augmentation.py

```diff
@@ -25,6 +25,334 @@
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(device).to(dtype), atol=1e-4, rtol=1e-4), output
+
+    def _test_random_p_1_return_transform_implementation(
+            self, device, dtype, input_tensor, expected_output, expected_transformation, **params):
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=True)
+        output, transformation = augmentation(input_tensor.to(device).to(dtype))
+
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(device).to(dtype), atol=1e-4, rtol=1e-4), output
+
+        # Transformation should match
+        assert transformation.shape == expected_transformation.shape
+        assert_allclose(transformation, expected_transformation.to(
+            device).to(dtype), atol=1e-4, rtol=1e-4), transformation
+
+    # TODO finish it
+
+    def _test_random_batch_transform_implementation(
+            self, device, dtype, input_tensor, expected_output, expected_transformation, batch_prob, **params):
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=True)
+        forward_params = {**augmentation.generate_parameters(input_tensor.shape), "batch_prob": batch_prob}
+        output, transformation = augmentation(input_tensor.to(device).to(dtype), params=forward_params)
+
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(device).to(dtype), atol=1e-4, rtol=1e-4), output
+
+        # Transformation should match
+        assert transformation.shape == expected_transformation.shape
+        assert_allclose(transformation, expected_transformation.to(
+            device).to(dtype), atol=1e-4, rtol=1e-4), transformation
+
+    # TODO finish it
+
+    def _test_sequential(self, device, dtype, **params):
+
+        f = nn.Sequential(
+            self._create_augmentation_from_params(**params, return_transform=True),
+            self._create_augmentation_from_params(**params, return_transform=True)
+        )
+
+        input = torch.rand(3, 5, 5).to(device)  # 3 x 5 x 5
+
+        expected = input
+
+        expected_transform = torch.eye(3).unsqueeze(0)  # 3 x 3
+        expected_transform = expected_transform.to(device)
+
+        assert_allclose(f(input)[0], expected, atol=1e-4, rtol=1e-5)
+        assert_allclose(f(input)[1], expected_transform)
+
+
+class TestColorJitterAlternative(CommonTests):
+
+    possible_params = {
+        "brightness": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "contrast": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "saturation": (1., (0., 2.), [0., 2.], torch.tensor(0.), torch.tensor((0., 2.))),
+        "hue": (0., (-0.5, 0.5), [-0.5, 0.5], torch.tensor(0.), torch.tensor((-0.5, 0.5))),
+    }
```

Comment:
I would maybe just leave default parameters to and let users to define possible parameters to each test

Reply:
Regarding this possible_params the idea was that I already bumped into situations when a theoretically valid argument set resulted in an exception. We could test this by parametrising the smoke tests to try every supported value for the single params. This possible_params dict is to put it into some ordered structure, that can be followed.

Reply:
I have fixed some in #714. I think we might need to have ```valid_input_params``` to test all combinations if possible.

Meanwhile, I am thinking of submiting another PR to add tests for ```K.augmentation.random_generators``` and ```K.augmentation.functionals```.